src/util/formatter.js

```diff
@@ -0,0 +1,80 @@
+import Beautify from 'js-beautify';
+
+const BEAUTIFY_SETTINGS = {
+  indent_size: 2,
+  indent_char: ' ',
+  indent_with_tabs: false,
+  eol: '\n',
+  end_with_newline: false,
+  indent_inner_html: true,
+  indent_level: 0,
+  preserve_newlines: false,
+  max_preserve_newlines: 10,
+  space_in_paren: false,
+  space_in_empty_paren: false,
+  jslint_happy: false,
+  space_after_anon_function: false,
+  brace_style: 'collapse',
+  unindent_chained_methods: false,
+  break_chained_methods: false,
+  keep_array_indentation: false,
+  unescape_strings: false,
+  wrap_line_length: 0,
+  e4x: false,
+  comma_first: false,
+  operator_position: 'before-newline',
+};
+
+function indexToCharIndex(string, index) {
```

Comment:
Implementation-wise, I think it would be simpler (and quite possibly more performant) to just write this as:

```js
string.slice(0, index).replace(/\s/, '').length
```